src/features/swap/index.tsx

```diff
@@ -274,6 +274,14 @@ const SwapWidget = ({ sell }: Params) => {
     }))
   }, [palette, darkMode, chainId])
 
+  useEffect(() => {
+    if (!sell) return
+    setParams((params) => ({
+      ...params,
+      sell,
+    }))
+  }, [sell])
+
```

Comment:
If the user is navigating from assets/dashboard to the swap page then this doesn't seem necessary. 
If however the user has copied the swap link with the asset it in it and now tries to load this page in a new tab then the asset is not properly set.